pkg/codegen/testing/test/testdata/internal-references-go/go/example/init.go

```diff
@@ -0,0 +1,42 @@
+// *** WARNING: this file was generated by test. ***
+// *** Do not edit by hand unless you're certain you know what you are doing! ***
+
+package example
+
+import (
+	"fmt"
+
+	"github.com/blang/semver"
+	"github.com/pulumi/pulumi-random/sdk/v4/go/random"
+	"github.com/pulumi/pulumi-tls/sdk/v4/go/tls"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+type pkg struct {
+	version semver.Version
+}
+
+func (p *pkg) Version() semver.Version {
+	return p.version
+}
+
+func (p *pkg) ConstructProvider(ctx *pulumi.Context, name, typ, urn string) (pulumi.ProviderResource, error) {
+	if typ != "pulumi:providers:example" {
+		return nil, fmt.Errorf("unknown provider type: %s", typ)
+	}
+
+	r := &Provider{}
+	err := ctx.RegisterResource(typ, name, nil, r, pulumi.URN_(urn))
+	return r, err
+}
+
+func init() {
+	var _ random.Provider
+	var _ tls.Provider
```

Comment:
These internal references are emitted because of the following in the schema:

```json
      "internalReferences": [
         "github.com/pulumi/pulumi-random/sdk/v4/go/random.Provider",
         "github.com/pulumi/pulumi-tls/sdk/v4/go/tls.Provider"
       ]
```